src/migrations/00084-split-device-metrics-from-device-schema.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE IF NOT EXISTS "device metrics record" (
+	"created at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+,	"modified at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL
+,	"id" SERIAL NOT NULL PRIMARY KEY
+,	"is reported by-device" INTEGER NOT NULL UNIQUE
+,	"memory usage" INTEGER NULL
+,	"memory total" INTEGER NULL
+,	"storage block device" VARCHAR(255) NULL
+,	"storage usage" INTEGER NULL
+,	"storage total" INTEGER NULL
+,	"cpu usage" INTEGER NULL
+,	"cpu temp" INTEGER NULL
+,	"is undervolted" BOOLEAN DEFAULT FALSE NOT NULL
+,	FOREIGN KEY ("is reported by-device") REFERENCES "device" ("id")
+);
+
+
+CREATE INDEX IF NOT EXISTS "device_metrics_record_by_device_idx"
+ON "device metrics record" ("is reported by-device");
```

Comment:
This needs to be included in `balena-init.sql`

Reply:
Thanks

Reply:
@Page- @fisehara Afaict we shouldn't need to add a manual index (neither here nor in the `balena-init.sql`), since the `"is reported by-device" INTEGER NOT NULL UNIQUE` part of the `CREATE TABLE` does auto-generate the following index:
```sql
CREATE UNIQUE INDEX "device metrics record_is reported by-device_key" ON public."device metrics record" USING ("is reported by-device");
```
which makes `device_metrics_record_by_device_idx` a duplicate index.

If we still prefer to have the index in the migration just in case, then it should probably be matching the auto-generated one.

Reply:
> Adding a unique constraint will automatically create a unique btree index on the column or group of columns used in the constraint.

See: https://www.postgresql.org/docs/current/sql-createtable.html